R/module_teal_with_splash.R

```diff
@@ -126,13 +126,16 @@ srv_teal_with_splash <- function(id, data, modules, filter = teal_slices()) {
         )
       )
 
-      validate(need(teal.data::datanames(data), "Data has no datanames. Contact app developer."))
-
+      if (!length(teal.data::datanames(data))) {
+        warning("`data` object has no datanames. Default datanames are set using `teal_data`'s environment.")
+      }
 
-      is_modules_ok <- check_modules_datanames(modules, teal.data::datanames(data))
-      validate(need(isTRUE(is_modules_ok), is_modules_ok))
+      is_modules_ok <- check_modules_datanames(modules, teal_data_datanames(data))
+      if (!isTRUE(is_modules_ok)) {
+        logger::log_warn(is_modules_ok)
```

Comment:
```suggestion
        logger::log_warn(is_modules_ok)
        validate(need(isTRUE(is_modules_ok), sprintf("%s. Contact app developer.", is_modules_ok)))
```

<img width="823" alt="Skärmavbild 2023-12-14 kl  13 11 39" src="https://github.com/insightsengineering/teal/assets/6959016/553a2ab5-e5ea-4f12-883e-1cb09b7cebdf">